### PR TITLE
OPCT-178: introduce job to validate upgrade mode and test 4.14

### PR DESCRIPTION
**Pull request by mtulio**
Introduce the periodic to validate upgrade mode (added on v0.3) with OPCT on vsphere provider using agnostic/None platform

https://issues.redhat.com/browse/OPCT-178

What's new?

- Added `platform-external-vsphere` periodic for 4.14
- Added `platform-none-vsphere` periodic for 4.13
- Added `upgrade` periodics for 4.11->4.12, and 4.12->4.13 [candidate] for `platform-none-vsphere`

Changes:

- Removed duplicated `openshift-provider-certification` from the job names
- Moved workflows to a prefix path of `opct`
- Bump Go build IS to 1.19
- Timeout is defined as 4h for regular, and 6h for upgrades
- Cron reviewed to run once a week by variant
- Review OWNERS, creating links to repeated files

OPCT features and OCP cluster matrix

| OPCT test/variant | Topology | OCP_4.10 |  OCP_4.11 |  OCP_4.12 | OCP_4.13 |  OCP_4.14 |
| -- | -- | -- | -- | -- | -- | -- |
| `platform-none-vsphere` | HighlyAvailable | OK/ stable | OK/ stable | OK/ stable | OK/ candidate | -- |
| `platform-none-vsphere-upgrade` | HighlyAvailable | -- | OK/ stable | OK/ candidate | -- |  -- |
| `platform-external-vsphere` | HighlyAvailable | --  | --  | -- | OK/ candidate | OK/ candidate |

Executions/rehearsals Dashboard:

- [platform-none-vsphere-4.10](https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/rehearse-38271-periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.10-platform-none-vsphere)
- platform-none-vsphere-4.11: NA
- [platform-none-vsphere-upgrade-4.11](https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/rehearse-38271-periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.11-platform-none-vsphere-upgrade)
- platform-none-vsphere-4.12: NA
- [platform-none-vsphere-upgrade-4.12](https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/rehearse-38271-periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.12-platform-none-vsphere-upgrade)
- [platform-none-vsphere-4.13](https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/rehearse-38271-periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-platform-none-vsphere)
- platform-external-vsphere-4.13: NA
- [platform-external-vsphere-4.14](https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/rehearse-38271-periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-external-vsphere)

### CODE REVIEW

**Comment by mtulio** [comment]
PR is still in dev to test the new periodic. Uncc for a while
/uncc @dgoodwin @rvanderp3

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.11-platform-none-vsphere-upgrade

**Comment by mtulio** [comment]
Forcing `jq` check to not fail before installing.

/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.11-platform-none-vsphere-upgrade

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.11-platform-none-vsphere-upgrade

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.11-platform-none-vsphere-upgrade

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.11-platform-none-vsphere-upgrade

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.11-platform-none-vsphere-upgrade

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.11-platform-none-vsphere-upgrade

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.11-platform-none-vsphere-upgrade

/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-external-vsphere

/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-platform-none-vsphere

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-platform-none-vsphere

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-external-vsphere

File: ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml [hunk 1]
```diff
@@ -194,8 +360,84 @@ periodics:
...
+    org: redhat-openshift-ecosystem
+    repo: provider-certification-tool
+  labels:
+    ci-operator.openshift.io/cloud: vsphere
+    ci-operator.openshift.io/cloud-cluster-profile: vsphere
+    ci-operator.openshift.io/variant: "4.13"
+    ci.openshift.io/generator: prowgen
+    job-release: "4.13"
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-platform-external-vsphere
```
Comment (mtulio) [inline ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml:430]: Missing reporter_config

File: ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml [hunk 2]
```diff
@@ -274,8 +519,85 @@ periodics:
...
+    org: redhat-openshift-ecosystem
+    repo: provider-certification-tool
+  labels:
+    ci-operator.openshift.io/cloud: vsphere
+    ci-operator.openshift.io/cloud-cluster-profile: vsphere
+    ci-operator.openshift.io/variant: "4.14"
+    ci.openshift.io/generator: prowgen
+    job-release: "4.14"
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.14-platform-external-vsphere
```
Comment (mtulio) [inline ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml:589]: Missing reporter_config

Do we need for 4.14?

File: ci-operator/step-registry/provider-certification-tool/results/provider-certification-tool-results-ref.yaml [hunk 3]
```diff
@@ -15,4 +15,4 @@ ref:
       cpu: 10m
       memory: 100Mi
   documentation: >-
-    This step executes the provider certification tool to certify the OpenShift cluster.
+    Steps to retrieve results from OPCT execution and publish it on persistent storage (S3).
```
Comment (mtulio) [inline ci-operator/step-registry/provider-certification-tool/results/provider-certification-tool-results-ref.yaml:18]: Review referencing single step

**Review by mtulio — COMMENTED** [review]
A few nit comments, rehearsals passed!

File: ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml (hunk 2 above)
Comment (mtulio) [reply ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml:589]: cc @elmiko

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.10-platform-none-vsphere

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.12-platform-none-vsphere

**Comment by mtulio** [comment]
> @mtulio: all tests **passed!**
> 
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openshift&repo=release&pr=38271). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:mtulio).

🎉 
Hey Joe, ptal? 
/assign @jcpowermac

**Comment by jcpowermac** [comment]
/lgtm
/approve
/pj-rehearse ack
/hold

@mtulio release the hold when you are ready to merge.

**Comment by mtulio** [comment]
> /lgtm
> /approve
> /pj-rehearse ack
> /hold
> 
> @mtulio release the hold when you are ready to merge. 

Thanks!

/unhold

**Pull request merged**